atd-vze/src/views/Users/UserForm.js

```diff
@@ -0,0 +1,289 @@
+    },
+  };
+
+  const [userFormData, setUserFormData] = useState(defaultFormData);
+  const [isFormDataLoaded, setIsFormDataLoaded] = useState(false);
+  const [isSubmissionError, setIsSubmissionError] = useState(false);
+  const [isFormSubmitted, setIsFormSubmitted] = useState(false);
+
+  const roles = [
+    { id: "itSupervisor", label: "IT Supervisor" },
+    { id: "admin", label: "Admin" },
+    { id: "editor", label: "Editor" },
+    { id: "readOnly", label: "Read-only" },
+  ];
+
+  // Fetch existing user data if editing
+  useEffect(() => {
+    if (type === "Edit") {
+      const endpoint = `${process.env.REACT_APP_CR3_API_DOMAIN}/user/get_user/${id}`;
+      axios
+        .get(endpoint, {
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        })
+        .then(res => {
+          setUserFormData(res.data);
+          setIsFormDataLoaded(true);
+        });
+    }
+  }, [id, type, token]);
+
+  const handleTextInputChange = event => {
+    const updatedFormData = {
+      ...userFormData,
+      [event.target.id]: event.target.value,
+    };
+    setUserFormData(updatedFormData);
+  };
+
+  const handleRoleRadioInputChange = event => {
+    const field = "app_metadata";
+    const appMetadata = {
+      roles: [event.target.value],
+    };
+    const updatedFormData = { ...userFormData, [field]: appMetadata };
+
+    setUserFormData(updatedFormData);
+  };
+
+  // Remove fields not needed for edits (so req does not fail)
+  const cleanFormDataForEdit = userFormData => {
+    const editFields = ["name", "email", "app_metadata"];
+
+    // If resetting password, must include password and required connection field
+    userFormData.password !== "" &&
+      editFields.push("password") &&
+      editFields.push("connection");
+
+    const cleanedFormData = editFields.reduce((acc, field) => {
+      return { ...acc, [field]: userFormData[field] };
+    }, {});
+
+    return cleanedFormData;
+  };
+
+  const handleFormSubmit = () => {
+    if (type === "Edit") {
+      const endpoint = `${process.env.REACT_APP_CR3_API_DOMAIN}/user/update_user/${id}`;
+      const updatedFormData = cleanFormDataForEdit(userFormData);
+      axios
+        .put(endpoint, updatedFormData, {
+          headers: { Authorization: `Bearer ${token}` },
+        })
+        .then(() => {
+          setIsFormSubmitted(true);
+        })
+        .catch(() => {
+          setIsSubmissionError(true);
+        });
+    } else if (type === "Add") {
+      const endpoint = `${process.env.REACT_APP_CR3_API_DOMAIN}/user/create_user`;
+      axios
+        .post(endpoint, userFormData, {
+          headers: { Authorization: `Bearer ${token}` },
+        })
+        .then(() => {
+          setIsFormSubmitted(true);
+        })
+        .catch(() => {
+          setIsSubmissionError(true);
+        });
+    }
+  };
+
+  const resetForm = () => {
+    setUserFormData(defaultFormData);
+  };
+
+  const renderErrorMessage = () => (
```

Comment:
this is smart to have helpful error messaging. 👏

Reply:
Auth0's API and documentation is really great. 👏 I'll have to keep this in mind for our tech debt sprint. 😄

Reply:
Just realized that this branch didn't incorporate the error responses (which are great) from the API here quite yet. 😆Stay tuned for that in a further branch!